src/scriptworker/cot/verify.py

```diff
@@ -1176,7 +1179,7 @@ async def _get_additional_github_pull_request_jsone_context(decision_link):
     github_repo = GitHubRepository(repo_owner, repo_name, token)
     repo_definition = github_repo.definition
 
-    if repo_definition["fork"]:
+    if repo_definition["fork"] and base_repo_url != repo_url:
```

Comment:
I _think_ this is fine. It looks like the original `github_repo` comes from `head_repository` (via `get_repo`). From there we have a few cases:
* Head `github_repo` is not a fork, and we keep it. Eg: opening a PR to `firefox-ios` from a branch in that repo. Obviously we need the original `github_repo` in this case (and redoing it would end up with the same data anyways...)
* Head `github_repo` is a fork, and the base repo is different. Eg: opening a PR to `firefox-ios` from `staging-firefox-ios`, or even `staging-firefox-ios` from some user's repository. In this case we must redefine `github_repo` again, because we can't trust whatever data comes out of the head `github_repo`. I guess we can trust whatever `repo_definition["parent"]` provides us because there's no way it could be anything _except_ the repo that the event is firing from?
* Head `github_repo` is a fork and the base repo is the same. Eg: opening a PR to `staging-firefox-ios` from another branch in that repository. We can use the existing head `github_repo` in this case because the only people that could open this PR would be those that can modify the one on `main` anyways? (I guess that's mostly true...depending on branch protections we could theoretically have a case where someone could push to a branch, open a PR, but be unable to modify things on `main` directly. I'm not sure if that's worth dealing with, and it's certainly not something to deal with here if so.)

Two suggestions:
* Add a comment that explains this stuff a bit - it's a little bit head spinning IMO (doesn't need to be my textwall :laughing:)
* Consider defaulting to the base `github_repo`. Obviously this doesn't change anything now, but it might protect against accidents in a future refactor.

Reply:
Comment added!

It occurred to me that we could have the case that `github_repo` is not a fork, but it still opens a pull request on an upstream repo. E.g, we open a pull request from `staging-xpi-manifest` to `xpi-manifest`. But in this case I don't think the parent repo information would be populated in the Github API return value.. so I'm not sure there's anything we can do about it unless we trust what's in the `BASE_REPOSITORY` env

Reply:
> Consider defaulting to the base github_repo. Obviously this doesn't change anything now, but it might protect against accidents in a future refactor.

Sorry, I'm confused by what you mean here

Reply:
> > Consider defaulting to the base github_repo. Obviously this doesn't change anything now, but it might protect against accidents in a future refactor.
> 
> Sorry, I'm confused by what you mean here

I guess what I'm trying to guard against is some future refactor that causes a bug where we don't set `github_repo` a second time, and we wrongly end up with the original `github_repo` definition. In my head when I wrote the earlier comment that might be a security issue...but looking at it again the worst that can happen is a failure to verify CoT....so I don't really care now!